editor/src/core/shared/element-template.ts

```diff
@@ -871,6 +871,53 @@ export function elementReferencesElsewhere(element: JSXElementChild): boolean {
   }
 }
 
+export function getElementReferencesElsewherePathsFromProps(
+  element: JSXElementChild,
+  pathSoFar: PropertyPath,
+): PropertyPath[] {
+  switch (element.type) {
+    case 'JSX_ELEMENT':
+      return element.props.flatMap((prop) =>
+        prop.type === 'JSX_ATTRIBUTES_SPREAD'
+          ? []
+          : getElementReferencesElsewherePathsFromProps(
+              prop.value,
+              PP.append(pathSoFar, PP.create(prop.key)),
+            ),
+      )
+    case 'ATTRIBUTE_NESTED_OBJECT':
+      const spreads: JSXSpreadAssignment[] = []
+      const assigments: JSXPropertyAssignment[] = []
+      element.content.forEach((c) =>
+        c.type === 'PROPERTY_ASSIGNMENT'
```

Comment:
I'd argue this would be clearer with regular if..else statements, as we're discarding the result.

Reply:
implemented in https://github.com/concrete-utopia/utopia/pull/3778/commits/92647e40a27037686556d8b42f8277f4e66eb548